### PR TITLE
refactor: always recommend `from_native`

### PR DESCRIPTION
**Pull request by EdAbati**
## What type of PR is this? (check all applicable)

- [x] 💾 Refactor
- [ ] ✨ Feature
- [ ] 🐛 Bug Fix
- [ ] 🔧 Optimization
- [ ] 📝 Documentation
- [ ] ✅ Test
- [ ] 🐳 Other

## Checklist

- [x] Code follows style guide (ruff)
- [x] Tests added 
- [x] Documented the changes

## If you have comments or can explain your changes, please do so below.

As implemented in #331, I like the idea of recommending to use `nw.from_native` instead of the classes directly. Moving this to a separate PR as @MarcoGorelli asked. 
The recommendation was already mentioned in some docstrings, but not applied everywhere. This also allows to simplify the `LazyFrame.__init__`.

### CODE REVIEW

File: tests/hypothesis/test_concat.py [hunk 1]
```diff
@@ -44,27 +44,29 @@ def test_concat(  # pragma: no cover
 
     if how == "horizontal":
         df_pl = (
-            nw.LazyFrame(df_polars)
+            nw.from_native(df_polars)
+            .lazy()
```
Comment (EdAbati) [inline tests/hypothesis/test_concat.py:48]: I kept this like it was #331 for now, but I'm not 100% sure why we are testing here (and other tests below) the `LazyFrame` interface, when both outputs are not lazy.

What would we miss it we just had `from_native()`?

Comment (MarcoGorelli) [reply tests/hypothesis/test_concat.py:48]: 😄 this looks like a mistake tbh, I think `from_native` should be fine, thanks for noticiing

Comment (MarcoGorelli) [reply tests/hypothesis/test_concat.py:48]: I _might_ have been chasing coverage with some of these...but this `.lazy().collect()` definitely feels useless

Comment (EdAbati) [reply tests/hypothesis/test_concat.py:48]: > I might have been chasing coverage with some of these

I'll try to delete some of them and see if coverage is happy 😅

**Comment by EdAbati** [comment]
Deleted some redundant `.lazy()` calls in test here: https://github.com/narwhals-dev/narwhals/pull/358/commits/a4ee7e9f0981db6fffe5dbb05f05476f9a32c76f

Coverage is 100% happy

**Review by MarcoGorelli — APPROVED** [review]
thanks @EdAbati !

**Pull request merged**